gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/RecommendedProductDependenciesExtension.java

```diff
@@ -47,6 +49,17 @@ public final void productDependency(@DelegatesTo(ProductDependency.class) Closur
         }));
     }
 
+    final Provider<Set<ProductDependency>> getFinalizedRecommendedProductDependencies() {
+        if (GradleVersion.current().compareTo(GradleVersion.version("5.0")) >= 0) {
+            return recommendedProductDependencies.map(rpd -> {
+                recommendedProductDependencies.finalizeValue();
```

Comment:
this stuff is actually not necessary anymore since we now guarantee that we don't read this until execution time.